pymc3/distributions/bound.py

```diff
@@ -14,218 +14,30 @@
-        self_repr = super()._str_repr(**kwargs)
-
-        if "formatting" in kwargs and "latex" in kwargs["formatting"]:
-            return self_repr + " -- " + distr_repr
-        else:
-            return self_repr + "-" + distr_repr
-
-
-class _DiscreteBounded(_Bounded, Discrete):
-    def __init__(self, distribution, lower, upper, transform="infer", *args, **kwargs):
-        if transform == "infer":
-            transform = None
-        if transform is not None:
-            raise ValueError("Can not transform discrete variable.")
-
-        if lower is None and upper is None:
-            default = None
-        elif lower is not None and upper is not None:
-            default = (lower + upper) // 2
-        if upper is not None:
-            default = upper - 1
-        if lower is not None:
-            default = lower + 1
-
-        super().__init__(distribution, lower, upper, default, *args, transform=transform, **kwargs)
-
-
-class _ContinuousBounded(_Bounded, Continuous):
-    r"""
-    An upper, lower or upper+lower bounded distribution
-
-    Parameters
-    ----------
-    distribution: pymc3 distribution
-        Distribution to be transformed into a bounded distribution
-    lower: float (optional)
-        Lower bound of the distribution, set to -inf to disable.
-    upper: float (optional)
-        Upper bound of the distribibution, set to inf to disable.
-    tranform: 'infer' or object
-        If 'infer', infers the right transform to apply from the supplied bounds.
-        If transform object, has to supply .forward() and .backward() methods.
-        See pymc3.distributions.transforms for more information.
-    """
-
-    def __init__(self, distribution, lower, upper, transform="infer", *args, **kwargs):
-        if lower is not None:
-            lower = at.as_tensor_variable(floatX(lower))
-        if upper is not None:
-            upper = at.as_tensor_variable(floatX(upper))
-
-        if transform == "infer":
-            if lower is None and upper is None:
-                transform = None
-                default = None
-            elif lower is not None and upper is not None:
-                transform = transforms.interval(lower, upper)
-                default = 0.5 * (lower + upper)
-            elif upper is not None:
-                transform = transforms.upperbound(upper)
-                default = upper - 1
-            else:
-                transform = transforms.lowerbound(lower)
-                default = lower + 1
-        else:
-            default = None
+class BoundRV(RandomVariable):
+    name = "bound"
+    ndim_supp = 0
+    ndims_params = [0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("Bound", "\\operatorname{Bound}")
 
-        super().__init__(distribution, lower, upper, default, *args, transform=transform, **kwargs)
+    @classmethod
+    def rng_fn(cls, rng, distribution, lower, upper, size):
+        raise NotImplementedError("Cannot sample from a bounded variable")
```

Comment:
@ricardoV94, This is what we want right?

Reply:
Yes. But it means you need to compute the `initval` instead, otherwise the model will raise an error when a bounded variable is created.